src/impls/$select-by/$select-by.js

```diff
@@ -0,0 +1,30 @@
+import { $async, $await } from '../../../generate/async.macro.cjs';
+
+import { $iterableCurry } from '../../internal/$iterable.js';
+import { $__peekerate } from '../$peekerate/$peekerate.js';
+
+$async;
+export function $__selectBy(iterable, selector, mapper) {
```

Comment:
I think you should make `mapper` optional, with a default of `value => value`.

Reply:
If we do that, it would make it equivalent to the select function, so it would be redundant. As mentioned in the comment, while we could have a single function with both mapper/selector optional, I think it would not play well with type inference/completion in the editor.

Reply:
Hmm, idunno. I think I'd have to play with the real type definitions and see if there was an acceptable compromise. My typedefs do contain quite a few compromises, especially around currying, due to limitations of Typescript. I'm generally fine with this as iter-tools is a JS library first and a TS library second.

I'm hoping that for iter-tools 8 I'll be able to write a type for my currying function and drop all the hacked up custom overloads I've written. TS has relatively recently gained the power to express  such a currying function, but it's still gonna be a bear to write that type, and I've no guarantee the resultant error messages will make sense to human beings.